web-app/frontend/src/instruments/helpers/AddNameManager.ts

```diff
@@ -65,6 +65,24 @@ export class AddNameManager {
       this.updateRemoveButtons(); // Ensure correct behavior when rows are removed
     });
 
+    // Add event listener for left/right direction and alignment of the name input based on language selection
+    const langInput = row.querySelector(
+      `#language${index}`,
+    ) as HTMLInputElement;
+    const nameInput = row.querySelector(`#name${index}`) as HTMLInputElement;
+
+    langInput.addEventListener('change', () => {
+      const lang = this.languages.find(
+        (l) => l.wikidata_code === langInput.value,
+      );
+
+      if (lang) {
+        nameInput.classList.remove('force-rtl', 'force-ltr');
```

Comment:
This is a weird place to clean up the class. You should always clean up when the language changes. Move this outside of `if` and up to make it the first thing to do when the change happens. Also remember to trigger event in `restoreFormData()` after `L307` to make sure the change event gets fired.